pkg/core/statesync/module.go

```diff
@@ -74,11 +90,37 @@ type Ledger interface {
 	HeaderHeight() uint32
 }
 
+// Checkpoint stores the state of an interrupted contract storage state sync in
+// ContractStorageBased mode.
+type Checkpoint struct {
+	// MptRoot is a computed intermediate MPT root at the Checkpoint.
+	MptRoot util.Uint256
+	// isSynced indicates whether sync process is completed.
+	isSynced bool
```

Comment:
// IsMPTSynced indicates whether MPT sync process is completed.
IsMPTSynced bool